src/main/java/net/openhft/chronicle/core/io/ReferenceCountedTracer.java

```diff
@@ -28,10 +28,7 @@ public interface ReferenceCountedTracer extends ReferenceCounted {
     @NotNull
     static ReferenceCountedTracer onReleased(final Runnable onRelease, Supplier<String> uniqueId, Class<?> type) {
         return Jvm.isResourceTracing()
-                ? new DualReferenceCounted(
-                new TracingReferenceCounted(onRelease, uniqueId.get(), type),
-                new VanillaReferenceCounted(() -> {
-                }, type))
+                ? new TracingReferenceCounted(onRelease, uniqueId.get(), type)
```

Comment:
Just use `TracingReferenceCounted` instead of dual

Reply:
As long as we are confident the tests are comprehensive enough, this is fine. 
What is the benefit of this change?

Reply:
See https://github.com/OpenHFT/Chronicle-Core/issues/445 (linked in the description & commit). DualReferenceCounted means event handlers e.g. the current `onRelease` handler and the proposed `ReferenceChangeListener` will be executed inside a synchronized block. We shouldn't execute arbitrary code while holding a resource as it introduces the risk of a deadlock. The first use of ReferenceChangeListener, triggering a cache flush in ReferenceCountedCache did deadlock when executed synchronously.